feat(serverless): 502 errors (#4170)
scaleway/docs-content: serverless/containers/troubleshooting/502-error-when-scaling.mdx

---
meta:
  title: Error code 502
  description: Troubleshoot 502 issues while sending requests to Serverless Containers.
content:
  h1: Error code 502
  paragraph: Troubleshoot 502 issues while sending requests to Serverless Containers.
tags: serverless Containers troubleshooting issue error 502
dates:
  validation: 2025-01-03
  posted: 2025-01-03
categories:
  - serverless
---

## Problem

> In some cases 502 errors can occur on Serverless Containers during autoscaling process: when the system creates new instances of the Container.

Reviewer: ```suggestion
502 errors can sometimes occur on Serverless Containers during the autoscaling process when new container instances are being created.
```

### Cause

> When the new instance is created, Serverless healthcheck verification is by default done on the HTTP port. Once it's opened, the ressource is considered ready and traffic starts to be forwarded to it.

Reviewer: ```suggestion
When a new instance is created, the Serverless health check verification is performed by default on the HTTP port. Once the port is opened, the resource is considered ready, and traffic starts being forwarded to it.
```

> This is commonly due to the traffic being sent too early to the newly created Container.

Reviewer: ```suggestion
This issue typically occurs when traffic is sent too early to the newly created container before it is fully deployed.
```

### Possible solutions

> To ensure the Serverless Container is completly ready, not only the HTTP port opened but also in the software scope the solution is to use **custom healthchecks**.

Reviewer: ```suggestion
To ensure the Serverless Container is completely ready, both in terms of the HTTP port and the software scope, it is recommended to use **custom health checks**.
```

> **Custom healthchecks** are available via the API, in this case it's recommended to implement a custom http route on the Container that returns the readiness status.

Reviewer: ```suggestion
**Custom health checks** can be configured via the API; in this case, it's recommended to implement a custom HTTP route on the Container that returns its readiness status.
```

> * [Link to API documentation](https://www.scaleway.com/en/developers/api/serverless-containers/#path-containers-update-an-existing-container): look for `health_check`.

Reviewer: ```suggestion
* [API Documentation for custom health checks](https://www.scaleway.com/en/developers/api/serverless-containers/#path-containers-update-an-existing-container) - Refer to the `health_check` parameter.
```

<Message type="tip">
  A simple health check to implement is just a simple `GET` on `/` to make sure the HTTP server is properly listening.

  Example (incomplete) settings:
  ```
  health_check: {
    http: {
        path: "/"
    }
    }
  ```
</Message>

<Message type="note">

>     Custom healthcheck feature will be available in the Scaleway console for Q1 2025.

Reviewer: ```suggestion
    The custom health check feature will be available in the Scaleway console in Q1 2025.
```
